addons/website_event/controllers/main.py

```diff
@@ -38,31 +38,27 @@ def sdn(date):
 
         def sd(date):
             return fields.Datetime.to_string(date)
+
+        def get_month_filter_domain(filter_name, filter_string, first_day, months_delta):
+            return [filter_name, filter_string, [
+                ("date_end", ">=", sd(first_day + relativedelta(months=months_delta))),
+                ("date_begin", "<", (first_day + relativedelta(months=months_delta+1)).strftime(datetime_format))],
+                0]
+
         today = datetime.today()
+        first_day_of_the_month = today.replace(day=1)
+        datetime_format = '%Y-%m-%d 00:00:00'
         dates = [
-            ['all', _('Next Events'), [("date_end", ">", sd(today))], 0],
+            ['all', _('Upcoming Events'), [("date_end", ">", sd(today))], 0],
             ['today', _('Today'), [
                 ("date_end", ">", sd(today)),
                 ("date_begin", "<", sdn(today))],
                 0],
-            ['week', _('This Week'), [
-                ("date_end", ">=", sd(today + relativedelta(days=-today.weekday()))),
-                ("date_begin", "<", sdn(today + relativedelta(days=6-today.weekday())))],
-                0],
-            ['nextweek', _('Next Week'), [
-                ("date_end", ">=", sd(today + relativedelta(days=7-today.weekday()))),
-                ("date_begin", "<", sdn(today + relativedelta(days=13-today.weekday())))],
-                0],
-            ['month', _('This month'), [
-                ("date_end", ">=", sd(today.replace(day=1))),
-                ("date_begin", "<", (today.replace(day=1) + relativedelta(months=1)).strftime('%Y-%m-%d 00:00:00'))],
-                0],
-            ['nextmonth', _('Next month'), [
-                ("date_end", ">=", sd(today.replace(day=1) + relativedelta(months=1))),
-                ("date_begin", "<", (today.replace(day=1) + relativedelta(months=2)).strftime('%Y-%m-%d 00:00:00'))],
-                0],
+            get_month_filter_domain('month', _('This month'), first_day_of_the_month, 0),
+            get_month_filter_domain('nextmonth1', _((today + relativedelta(months=1)).strftime('%B')), first_day_of_the_month, 1),
+            get_month_filter_domain('nextmonth2', _((today + relativedelta(months=2)).strftime('%B')), first_day_of_the_month, 2),
```

Comment:
This translation thing will most likely not work.
Maybe `.strftime('%B')` is magically localized?

To test and 🤞

Reply:
A priori, 'This month' is translated into 'Ce mois'. But March is still 'March'
If 'March' had a translation, I guess this will work. (but it's weird that it doesn't)

Reply:
```python
>>> from babel.dates import format_date
>>> format_date(date.today(), 'LLLL', locale='fr_FR')                                                                                                                                             
'février'
````

Reply:
You are an amazing working pinguin and a great coworker @mart-e

Reply:
Well actually, it is even better to use `odoo.tools.misc.format_date`. But your are an amazing coworker too, stay 100% yourself !

Reply:
Do not worry, oh mighty @mart-e, I got the idea to use our wonderful framework that directly integrates babel. So much power in Odoo !